continution to #1708 (#1963)
Codecademy/docs: content/go/concepts/interfaces/interfaces.md

    dating
    love()
    arranged()
}

type dating interface {
    compatibility()
}

type john struct {
    age int
}

func (a john) love() {
    fmt.Println("love with john")
}

func (a john) arranged() {
    fmt.Println("arranged marriage with john")
}
func (a john) compatibility() {
    fmt.Println("compatible with john")
}

type parents struct {
    //Named marriage interface
    m     marriage
    quote string
}

type lovers struct {
    // Unnamed marriage interface
    marriage
    quote string
}

func main() {
    a := john{age: 24}
    p := parents{quote: "I saw that you were perfect, and so I loved you. Then I saw that you were not perfect and I loved you even more", m: a}

    // p.love()
    // As this has named marriage interface m calling disquarely with .m would raise compiler error
    /*-----------------------ERROR-----*---------------------------*/
    // ./prog.go:39:4: p.love undefined (type parents has no field or method love)

    p.m.love()
    p.m.compatibility()
    p.m.arranged()

    l := lovers{quote: "I saw that you were perfect, and so I loved you. Then I saw that you were not perfect and I loved you even more", marriage: a}

>     fmt.Println("\n", l.quote, "\n")

Reviewer: ```suggestion
    fmt.Println("\n", l.quote)
```

    l.love()
    l.compatibility()
    l.arranged()
}
```

This example results in the following output:

```shell
love with john
compatible with john
arranged marriage with john

I saw that you were perfect, and so I loved you. Then I saw that you were not perfect and I loved you even more

love with john
compatible with john
arranged marriage with john

Program exited.
```

## Codebyte Example

A simple example on how to use interfaces, `square` and `circle` both implement the `geometry` interface as both implement the required methods.

In the code below, the `geometry` interface has two methods `area()` and `perimeter()`, both of which return a float64. `square` and `circle` implement these methods, allowing them to be used interchangeably as geometric objects.

The code then defines a `printGeometry()` function, which takes a geometry object as its parameter and prints its details, including its area and perimeter.

In the `main()` function, a square and a circle are created, and the `printGeometry()` function is called for each of them, demonstrating how the two different types can be used interchangeably as geometric objects.

```codebyte/golang

> 

Reviewer: ```suggestion
```suggestion
```
```

package main

import (
    "fmt"
    "math"
)

type geometry interface {
    area() float64
    perimeter() float64
}

type square struct {
    side float64
}
type circle struct {
    radius float64
}

func (s square) area() float64 {
    return s.side*s.side
}
func (s square) perimeter() float64 {
    return 4*s.side
}

func (c circle) area() float64 {
    return math.Pi * c.radius * c.radius
}
func (c circle) perimeter() float64 {
    return 2 * math.Pi * c.radius
}

func printGeometry(g geometry) {
    fmt.Println(g)
    fmt.Println(g.area())
    fmt.Println(g.perimeter())
}

func main() {
    s := square{side: 4}
    c := circle{radius: 5}

    printGeometry(s)
    printGeometry(c)
}

> 

Reviewer: ```suggestion
```

    p := parents{quote: "I saw that you were perfect, and so I loved you. Then I saw that you were not perfect and I loved you even more", m: a}

    // p.love()
    // As this has named marriage interface m calling disquarely with .m would raise compiler error
    /*-----------------------ERROR-----*---------------------------*/
    // ./prog.go:39:4: p.love undefined (type parents has no field or method love)

    p.m.love()
    p.m.compatibility()
    p.m.arranged()

    l := lovers{quote: "I saw that you were perfect, and so I loved you. Then I saw that you were not perfect and I loved you even more", marriage: a}
    fmt.Println("\n", l.quote, "\n")
    // It's possible to disquarely access the methods of embedded interface if the embedded interface is anonymous or unnamed.
    l.love()
    l.compatibility()
    l.arranged()
}
```

This example results in the following output:

```shell
love with john
compatible with john
arranged marriage with john

I saw that you were perfect, and so I loved you. Then I saw that you were not perfect and I loved you even more

love with john
compatible with john
arranged marriage with john

Program exited.
```

## Codebyte Example

> A simple example on how to use interfaces, Square and circle both implement geometry interface as both square and circle implements area and perimeter methods

Reviewer: ```suggestion
A simple example on how to use interfaces, `square` and `circle` both implement the `geometry` interface as both implement the required methods.
```

> 
> <h3>Explanation<h3>

Reviewer: ```suggestion
```

> In the Below Code The geometry interface has two methods area() and perimeter(), both of which return a float64. The square and circle structs implement these methods, allowing them to be used interchangeably as geometry objects.
> 
> The code then defines a printGeometry() function, which takes a geometry object as its parameter and prints its details, including its area and perimeter.
> 
> In the main() function, a square and a circle are created, and the printGeometry() function is called for each of them, demonstrating how the two different types can be used interchangeably as geometry objects.

Reviewer: ```suggestion
In the code below, the `geometry` interface has two methods `area()` and `perimeter()`, both of which return a float64. `square` and `circle` implement these methods, allowing them to be used interchangeably as geometric objects.

The code then defines a `printGeometry()` function, which takes a geometry object as its parameter and prints its details, including its area and perimeter.

In the `main()` function, a square and a circle are created, and the `printGeometry()` function is called for each of them, demonstrating how the two different types can be used interchangeably as geometric objects.
```
